_rules/image-accessible-name-descriptive-qt1vmo.md

```diff
@@ -0,0 +1,295 @@
+---
+id: qt1vmo
+name: Image accessible name is descriptive
+rule_type: atomic
+description: |
+  This rule checks that images with an accessible name describe their topic or purpose.
+accessibility_requirements:
+  wcag20:1.1.1: # Non-Text Content (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag-technique:G94: # Providing short text alternative for non-text content that serves the same purpose and presents the same information as the non-text content
+    forConformance: false
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag-technique:G95: # Providing short text alternatives that provide a brief description of the non-text content
+    forConformance: false
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - Accessibility Tree
+  - DOM Tree
+  - CSS Styling
+  - Language
+acknowledgments:
+  authors:
+    - Wilco Fiers
+  assets:
+    - W3C; HTML and W3C logo
+    - Wikimedia; Fireworks image
+    - Adobe; PDF logo
+htmlHintIgnore:
+  # https://www.npmjs.com/package/htmlhint
+  # (used with `npm test` to ensure validity of code snippets)
+  - 'alt-require'
+  - 'attr-lowercase'
+---
+
+## Applicability
+
+Any `img`, `canvas` or `svg` element that is [visible][] and has a non-empty [accessible name][], except if one of the following is true:
+
+- The element has an [ancestor][] in the [flat tree][] that is [named from author][]; or
+- The element is an `img` element where the [current request][]'s [state][image request state] is not [completely available][].
+
+## Expectation
+
+Each test target has an [accessible name][] that serves an equivalent purpose to the [non-text content][] of that test target.
+
+## Assumptions
+
+This rule assumes that the language of each test target can be correctly determined (either programmatically or by analyzing the content).
```

Comment:
Not really sure that this is needed. The rule checks that the accessible name serves an equivalent purpose to the non-text content. How is identifying the language related?

Also, technically, you want to determine the language of the accessible name, not of the test target (which is an image, and, unless it's an image of text, won't have a language)

Reply:
> Not really sure that this is needed. The rule checks that the accessible name serves an equivalent purpose to the non-text content. How is identifying the language related?

`<img src="bread.jpg" aria-label="pain" />` may be good or bad depending whether the language is French or English.